### PR TITLE
Add basic Chip pattern

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Added the default Chip component.
Added styling to be ported to Vanilla framework

## QA
- Run `./run`
- Go to http://localhost:8403/?path=/story/chip--default-story
- See it looks :ok_hand: 

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1332

### CODE REVIEW

File: src/components/Chip/Chip.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import "./Chip.scss";
+
+const Chip = ({ value }) => {
+  return (
+    <div className="p-chip">
```
Comment (barrymcgee) [inline src/components/Chip/Chip.js:8]: Should this be a button?

File: src/components/Chip/Chip.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  // XXX Ant - 22.06.20: hardcoding @extend %small-text; due to:
+  // https://github.com/canonical-web-and-design/vanilla-framework/issues/3133
+  font-size: .875rem;
+  line-height: 1rem;
+
+  background-color: rgba(0, 0, 0, 0.05);
+  border-radius: 1rem;
+  cursor: pointer;
+  display: inline-block;
+  margin-bottom: $sp-unit + $spv-inner--small - 2 * map-get($nudges, nudge--small);
```
Comment (barrymcgee) [inline src/components/Chip/Chip.scss:13]: 🤯

File: src/components/Chip/Chip.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
+@import "~vanilla-framework/scss/settings_spacing";
+
+.p-chip {
+  // XXX Ant - 22.06.20: hardcoding @extend %small-text; due to:
+  // https://github.com/canonical-web-and-design/vanilla-framework/issues/3133
+  font-size: .875rem;
+  line-height: 1rem;
+
+  background-color: rgba(0, 0, 0, 0.05);
```
Comment (barrymcgee) [inline src/components/Chip/Chip.scss:9]:
```suggestion
  background-color: rgba($color-x-dark, 0.05);
```

File: src/components/Chip/Chip.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  border-radius: 1rem;
+  cursor: pointer;
```
Comment (barrymcgee) [inline src/components/Chip/Chip.scss:11]: This should be conditional on the ability to click and something happen.

File: src/components/Chip/Chip.scss [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  display: inline-block;
+  margin-bottom: $sp-unit + $spv-inner--small - 2 * map-get($nudges, nudge--small);
+  margin-right: $sph-inner--small;
+  padding: 0 $sph-inner map-get($nudges, nudge--small);
+  white-space: nowrap;
+  user-select: none;
+
+  &:hover {
+    background-color: rgba(0, 0, 0, 0.08);
```
Comment (barrymcgee) [inline src/components/Chip/Chip.scss:20]:
```suggestion
    background-color: rgba($color-x-dark, 0.08);
```

File: src/components/Chip/Chip.scss [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  &.is-selected {
+    background-color: rgba(0, 0, 0, 0.2);
```
Comment (barrymcgee) [inline src/components/Chip/Chip.scss:24]:
```suggestion
    background-color: rgba($color-x-dark, 0.2);
```

File: src/components/Chip/Chip.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    &:hover {
+      background-color: rgba(0, 0, 0, 0.23);
```
Comment (barrymcgee) [inline src/components/Chip/Chip.scss:27]:
```suggestion
      background-color: rgba($color-x-dark, 0.23);
```

File: src/components/Chip/Chip.scss [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  }
+
+  &:active {
+    background-color: rgba(0, 0, 0, 0.2);
```
Comment (barrymcgee) [inline src/components/Chip/Chip.scss:32]:
```suggestion
    background-color: rgba($color-x-dark, 0.2);
```

File: src/components/Chip/Chip.scss (hunk 2 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss:13]: I know, I hate it too but copied from Vanilla.

File: src/components/Chip/Chip.scss (hunk 3 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss:9]: Can you put hex value in RGB?

File: src/components/Chip/Chip.scss (hunk 4 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss:11]: I should remove this as the chip in this state in not clickable.

**Comment by anthonydillon** [comment]
@barrymcgee thanks replied to them and will action them on the morning. Could you change the labels please to see the state please.

File: src/components/Chip/Chip.scss (hunk 3 above)
Comment (barrymcgee) [reply src/components/Chip/Chip.scss:9]: Yep, Sass will do the magic 🧙🏻‍♂️

File: src/components/Chip/Chip.scss (hunk 2 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss]: Removed as 0 is much more atomic.

File: src/components/Chip/Chip.scss (hunk 3 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss]: Done

File: src/components/Chip/Chip.scss (hunk 4 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss]: Removed

File: src/components/Chip/Chip.scss (hunk 5 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss]: Done

File: src/components/Chip/Chip.scss (hunk 6 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss]: Done

File: src/components/Chip/Chip.scss (hunk 7 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss]: Done

File: src/components/Chip/Chip.scss (hunk 8 above)
Comment (anthonydillon) [reply src/components/Chip/Chip.scss]: Done

**Comment by anthonydillon** [comment]
@barrymcgee updated, ready for another review.

**Pull request merged**